First  - Automated Project Setup Script

venkatsunilm opened the pull request:
- Added a firsttime_setup_project.sh script for one-time project setup.
- Cleaned up unnecessary node_modules from the project root as it might not be necessary. So diff can look big, 
- The script checks for Python 3.12+ and Node.js 18+.
- Installs frontend dependencies in the frontend directory.
- Sets up backend Python virtual environment in backend/.venv (creates if missing).
- Activates the backend venv and installs uv if not present.
- Syncs backend dependencies using uv and pyproject.toml.
- Builds the backend using the npm script.
- Prints clear next steps for running backend and frontend servers.
- Added a .gitattributes file to enforce consistent LF (Unix) line endings across the project.
- Normalized line endings in all source files to prevent cross-platform issues when working between Windows and WSL/Linux.
- Modified SpotWise slogan text from Perfect to Best, as it looks overtpromising

How to use:
chmod +x helper/scripts/firsttime_setup_project.sh
./helper/scripts/firsttime_setup_project.sh

Note:
- Users must activate the backend venv manually before running backend commands. Please do follow the main README. 
- Node.js and Python version checks help prevent common environment issues.

venkatsunilm committed:
Update .gitattributes to enforce LF line endings

venkatsunilm committed:
Normalize all line endings with updated .gitattributes

DavidRamosSal commented:
Cool, everything seems to work well. But regarding the Python changes:

-  what is your reasoning behind requiring the user to run the virtual environment manually in an automated script?
- uv can not only manage the python packages but also the python versions, so no need for pyenv

We can discuss these things later but I would prefer if we don't touch this repo/branch and we start working on the MVP on a new repo/branch from scratch. What do you think?

venkatsunilm commented:
@DavidRamosSal Good points and happy to know that it is working well :)

1. I ran into trouble using WSL, and after looking into it, I found that activating the virtual environment inside an npm script isn’t reliable across all systems. So, it’s seems the best and recommended way is to activate it manually before running backend commands.

2. That’s correct, UV supports both Python version management and dependency management, making pyenv unnecessary for most workflows. I have now removed and updated all the files relevant. Latest changes are in this "develop" branch now as first commit.

I agree with your approach and happy to follow. That’s also why I’ve requested admin permissions, to help set up branch protection rules on main and enforce PR restrictions rules, since I noticed it was possible to merge directly into main. Creating a dedicated develop branch for our work will keep things safer and more organized moving forward. We can work on individual feature branches and merge to develop branch. 

Optional, maybe in future we will not have any branches but only main, with strong rules and we can follow Trunk-Based Development approach if you are interested, but lets see, not a priority.
